services/provider/server/server.go

```diff
@@ -779,3 +779,45 @@ func (s *OCSProviderServer) PeerBlockPool(ctx context.Context, req *pb.PeerBlock
 	}
 	return &pb.PeerBlockPoolResponse{}, nil
 }
+
+// RevokeBlockPoolPeering RPC call to delete the bootstrap secret to stop peering
+func (s *OCSProviderServer) RevokeBlockPoolPeering(ctx context.Context, req *pb.RevokeBlockPoolPeeringRequest) (*pb.RevokeBlockPoolPeeringResponse, error) {
+
+	klog.Infof("RevokeBlockPoolPeering request received for CephBlockPool %s and bootstrap secret %s", req.Pool, req.SecretName)
+
+	cephBlockPool, err := s.cephBlockPoolManager.GetBlockPoolByName(ctx, string(req.Pool))
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Failed to find CephBlockPool resource %s: %v", req.Pool, err)
+	}
+
+	// delete secret and unset ref on the blockPool
+	err = s.cephBlockPoolManager.UnSetBootstrapSecretRef(ctx, req.SecretName, cephBlockPool)
+	// there might be a case where the bootstrap secret was deleted but request failed after this and there was a retry,
+	// if error is IsNotFound, that means it is safe to proceed as we have deleted the bootstrap secret
+	if err != nil && !kerrors.IsNotFound(err) {
```

Comment:
better handle the notfound on delete inside the func itself (same for the rest). Current func is doing two things, maybe state the same in the func name?